src/librbd/image/RemoveRequest.h

```diff
@@ -27,6 +27,61 @@ class RemoveRequest {
   void send();
 
 private:
+  /**
+   * @verbatim
+   *
+   *                                  <start>
+   *                                     |
+   *                                     v                      
+   *                                OPEN IMAGE-------------------
+   *                                     |                      | 
+   *                                     v                      |
+   * (error: bottom up)           CHECK EXCLUSIVE LOCK----      |
+   *  _______<_______                    |               |      |
+   * |               |                   v            (aquired) |
+   * |               |            AQUIRE EXCLUSIVE LOCK  |      |
+   * |               |               /   |               |      |
+   * |               |------<-------/    v               |      |
+   * |               |            VALIDATE IMAGE REMOVAL<-      |
+   * |               |                   |                      v
+   * |               |                   v                      |
+   * |               |               CHECK SNAPS                |
```

Comment:
Nit: the "check XYZ" actions are not really async states and can probably be left off the chart